src/VisualStudio/Core/Def/Packaging/IPackageSearchIOService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServices.Packaging
+{
+    /// <summary>
+    /// Used so we can mock out how the search service does IO for testing purposes.
+    /// </summary>
+    internal interface IPackageSearchIOService
+    {
+        void Create(DirectoryInfo directory);
+        void Delete(FileInfo file);
+        IEnumerable<FileInfo> EnumerateFiles(DirectoryInfo directory);
+        bool Exists(FileSystemInfo info);
```

Comment:
In hte absense of "TryXXX" i'm definitely going to go with the approach of ```if (Exists()) { try { SomeIO() } catch ...```.  It is the best compromise AFAICT with the current BCL.  It avoids exceptions in the absence of races, but also acts correctly in the presence of them.